src/service.js

```diff
@@ -177,7 +190,18 @@ class Service {
   }
 
   patch (id, data, params) {
-    params.query = params.query || {};
+    const query = params.query || {};
+    const patchQuery = {};
+
+    // Account for potentially modified data
```

Comment:
We could add a flag for that case. But in order to get real-time events there isn't really a way around returning all changed items. I think it makes sense to change the adapters to what I suggested because then it'll at least return only the changed items (right now it returns everything that matches the query).